pkg/dyninst/compiler/snapshot_test.go

```diff
@@ -0,0 +1,88 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package compiler
+
+import (
+	"bytes"
+	"flag"
+	"os"
+	"path"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/pkg/dyninst/compiler/codegen"
+	"github.com/DataDog/datadog-agent/pkg/dyninst/compiler/sm"
+	"github.com/DataDog/datadog-agent/pkg/dyninst/irgen"
+	"github.com/DataDog/datadog-agent/pkg/dyninst/object"
+	"github.com/DataDog/datadog-agent/pkg/dyninst/testprogs"
+	"github.com/DataDog/datadog-agent/pkg/util/safeelf"
+)
+
+var rewriteFromEnv = func() bool {
+	rewrite, _ := strconv.ParseBool(os.Getenv("REWRITE"))
+	return rewrite
+}()
+var rewrite = flag.Bool("rewrite", rewriteFromEnv, "rewrite the test files")
```

Comment:
I'm surprised there isn't some existing machinery for snapshots, golden files, and rewrites already?